### PR TITLE
Jlc/match attributes of relationships for username

### PR DESCRIPTION
**Pull request by johanv26**
# Description
This configure the read of relationships for the api for each item.
For example now, username is a field that belongs to author and not
to the item of feedback in the api.
## After
Component of card continue working with username
Peek 2023-07-10 16-19

### CODE REVIEW

File: src/components/Feedback/FeedbackCarousel.tsx [hunk 1]
```diff
@@ -13,12 +13,31 @@ const { Carousel } = require('@edx/paragon');
...
     feedback: string,
+  },
+  relationships: {
+    author: {
+      data: Author
+    },
   }
 }
 
+interface JsonApiItem {
```
Comment (andrey-canon) [inline src/components/Feedback/FeedbackCarousel.tsx:26]: what is the purpose of this, since you are no reusing this ?

**Review by andrey-canon — CHANGES_REQUESTED** [review]
Tests

File: src/components/Feedback/FeedbackCarousel.tsx (hunk 1 above)
Comment (johanv26) [reply src/components/Feedback/FeedbackCarousel.tsx:26]: I started with this approach to include course_id, but the course_id would be another approach.
https://github.com/nelc/frontend-essentials/pull/26/commits/e2fcb24da4d44cd759ea68391a32e42300840061

File: src/components/Feedback/FeedbackCarousel.tsx [hunk 2]
```diff
@@ -13,9 +13,27 @@ const { Carousel } = require('@edx/paragon');
...
+      data: Author
+    },
+  }
+}
+
+interface JsonApiItem {
+  id: number,
+  type: string,
+  attributes: {
+    [key: string | number]: string | number
```
Comment (andrey-canon) [inline src/components/Feedback/FeedbackCarousel.tsx:30]: a key could be a number ? which is the case ?
I think you want to cover  multiples cases but it's still very specific,I mean you are not covering when the value is a boolean so do you need that it's not enough `attributes: object` ?

Comment (andrey-canon) [reply src/components/Feedback/FeedbackCarousel.tsx]: I think that's ok but I'm still wondering if `attributes: object` is not the same

Comment (johanv26) [reply src/components/Feedback/FeedbackCarousel.tsx]: [4bbd864](https://github.com/nelc/frontend-essentials/pull/26/commits/4bbd86464d122dfbf5e09898867d918b874a67a5)

Comment (andrey-canon) [reply src/components/Feedback/FeedbackCarousel.tsx]: That means that is the same  ?

Comment (johanv26) [reply src/components/Feedback/FeedbackCarousel.tsx]: Not the same actual object forced to not be a primitive.
So you cant use attributes: true, or attributes: 4, or attributes: "hi"
Also, there is an Object with `O` but this one allow anyone seems
https://stackoverflow.com/questions/49464634/difference-between-object-and-object-in-typescript
https://www.typescripttutorial.net/typescript-tutorial/typescript-object-type/

**Review by andrey-canon — APPROVED** [review]

**Pull request merged**